### PR TITLE
(DOCSP-27004): @realm/react-ify Custom User Data page

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

@realm/react-ify Custom User Data page

### Jira

- https://jira.mongodb.org/browse/DOCSP-27004

### Staged Changes

- [Custom User Data (React Native)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27004/sdk/react-native/manage-users/custom-user-data/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by cbullinger — APPROVED** [review]
overall lgtm!

File: examples/react-native/__tests__/ts/authentication/write-custom-user-data.test.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    const filter = {
+      userId: user.id, // Query for the user object of the logged in user
+    };
+    const updateDoc = {
+      $set: {
+        // Set User ID if it's not already set
+        userId: user.id,
+        // Set the logged in user's favorite color
+        favoriteColor,
```
[the remark this answers is not in the document]
Comment (mongodben) [reply examples/react-native/__tests__/ts/authentication/write-custom-user-data.test.tsx:60]: this is a javascript [shorthand property](https://ui.dev/shorthand-properties). this is equivalent to:

```js
favoriteColor: favoriteColor
```

it's a very common JS pattern.

File: source/sdk/react-native/manage-users/custom-user-data.txt [hunk 2]
```diff
@@ -10,164 +10,48 @@ Custom User Data - React Native SDK
...
+.. _react-native-write-custom-user-data:
 
-.. include:: /includes/custom-data-may-be-stale.rst
+Write Custom User Data
+----------------------
 
-If you have updated your custom user data within the last 30 minutes,
-use the :js-sdk:`refreshCustomData()
-<Realm.User.html#refreshCustomData>` method on the user object.
+You can write to a user's custom user data with :ref:`MongoDB Data Access <react-native-mongodb-data-access>`.
```
Comment (takameyer) [inline source/sdk/react-native/manage-users/custom-user-data.txt:47]: Is "MongoDB Data Access" a product?  Just thinking "with the MongoDB client" would sound a bit clearer.

**Review by takameyer — APPROVED** [review]
LGTM.  But we might want to revisit typing the user object.

File: source/sdk/react-native/manage-users/custom-user-data.txt (hunk 2 above)
Comment (mongodben) [reply source/sdk/react-native/manage-users/custom-user-data.txt:47]: yeah that's what we call it

**Comment by mongodben** [comment]
> LGTM. But we might want to revisit typing the user object.

what do you mean by 'typing the user object'?

**Pull request merged**